release_notes.py

```diff
@@ -632,41 +632,45 @@ def main():
-    modified_files, added_files, removed_files = filter_packagify_changes(modified_files, added_files,
-                                                                          removed_files, tag=tag)
-
-    for file_path in added_files:
-        create_file_release_notes('A', file_path)
-
-    for file_path in modified_files:
-        create_file_release_notes('M', file_path)
-
-    for file_path in removed_files:
-        handle_deleted_file(file_path, tag)
-
-    # join all release notes
-    res = []
-    beta_res = []
-    missing_release_notes = False
-    for key in RELEASE_NOTES_ORDER:
-        value = RELEASE_NOTE_GENERATOR[key]
-        ans, beta_ans = value.generate_release_notes(args.server_version)
-        if ans is None or value.is_missing_release_notes:
-            missing_release_notes = True
-        if ans:
-            res.append(ans)
-        if beta_ans:
-            beta_res.append(beta_ans)
-
-    release_notes = "\n---\n".join(res)
-    beta_release_notes = "\n---\n".join(beta_res)
-    create_content_descriptor(args.version, args.asset_id, release_notes, args.github_token, beta_rn=beta_release_notes)
-
-    if missing_release_notes:
-        print_error("Error: some release notes are missing. See previous errors.")
-        sys.exit(1)
+    try:
+        change_log = run_command('git diff --name-status {}'.format(args.git_sha1), exit_on_error=False)
+        modified_files, added_files, removed_files, _ = file_validator.get_modified_files(change_log)
+        modified_files, added_files, removed_files = filter_packagify_changes(modified_files, added_files,
+                                                                              removed_files, tag=tag)
+
+        for file_path in added_files:
+            create_file_release_notes('A', file_path)
+
+        for file_path in modified_files:
+            create_file_release_notes('M', file_path)
+
+        for file_path in removed_files:
+            handle_deleted_file(file_path, tag)
+
+        # join all release notes
+        res = []
+        beta_res = []
+        missing_release_notes = False
+        for key in RELEASE_NOTES_ORDER:
+            value = RELEASE_NOTE_GENERATOR[key]
+            ans, beta_ans = value.generate_release_notes(args.server_version)
+            if ans is None or value.is_missing_release_notes:
+                missing_release_notes = True
+            if ans:
+                res.append(ans)
+            if beta_ans:
+                beta_res.append(beta_ans)
+
+        release_notes = "\n---\n".join(res)
+        beta_release_notes = "\n---\n".join(beta_res)
+        create_content_descriptor(args.version, args.asset_id, release_notes, args.github_token, beta_rn=beta_release_notes)
+
+        if missing_release_notes:
+            print_error("Error: some release notes are missing. See previous errors.")
+            sys.exit(1)
+    except RuntimeError:
```

Comment:
RuntimeError is too generic. We can miss other exceptions here.
which line exactly causes this exception/error?

Reply:
@anara123 it shouldn't be too generic as it uses the exception that is being thrown in `run_command` function, which is called in line 636